test-suite/js/src/test/scala/org/scalajs/testsuite/library/FinalizationRegistryTest.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.library
+
+import org.junit.Test
+
+import scala.scalajs.js
+
+class FinalizationRegistryTest {
+  @Test def testMethods(): Unit = {
+    val registry = new js.FinalizationRegistry[js.Date, String, Any]((heldValue: String) => ())
+
+    val obj1 = new js.Date()
+    registry.register(obj1, "foo")
+
+    val obj2 = new js.Date()
+    registry.register(obj2, "bar", obj1)
```

Comment:
Did you mean
```suggestion
    registry.register(obj2, "bar", obj2)
```
?

Reply:
Fixed in https://github.com/scala-js/scala-js/pull/4366/commits/b226c06a9b417fae96da9aad4de173b40a50bdc9